### PR TITLE
Move the tag generation to a composite action

### PR DESCRIPTION
**Pull request by johnbillion**

### CODE REVIEW

File: .github/actions/prepare-tag-name/action.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  steps:
+    - name: Prepare tag name
+      shell: bash
+      run: | #shell
+        export LONG_TAG="${{ inputs.tag }}"
+        export SHORT_TAG="${LONG_TAG%.0}"
+        export BRANCH_NUMBER="${LONG_TAG%.*}"
+        echo short="$SHORT_TAG" >> "$GITHUB_OUTPUT"
+        echo long="$LONG_TAG" >> "$GITHUB_OUTPUT"
+        echo branch="$BRANCH_NUMBER" >> "$GITHUB_OUTPUT"
```
Comment (szepeviktor) [inline .github/actions/prepare-tag-name/action.yml]:
```suggestion
        LONG_TAG="${{ inputs.tag }}"
        {
          echo "short=${LONG_TAG%.0}"
          echo "long=${LONG_TAG}"
          echo "branch=${LONG_TAG%.*}"
        } >> "$GITHUB_OUTPUT"
```

Comment (szepeviktor) [reply .github/actions/prepare-tag-name/action.yml]: From #17
But this does not look uniform, it is not script-like.

Comment (johnbillion) [reply .github/actions/prepare-tag-name/action.yml]: What's the advantage of this?

Comment (szepeviktor) [reply .github/actions/prepare-tag-name/action.yml]: - no need for export
- does not look like copy-paste
- only one redirect to GITHUB_OUTPUT
- only one variable

Comment (szepeviktor) [reply .github/actions/prepare-tag-name/action.yml]: But it hurts the brain! 👺

Comment (johnbillion) [reply .github/actions/prepare-tag-name/action.yml]: Yeah it's a bit weird isn't it. I think I'll get rid of the exports and leave it at that. I need to remove direct use of the GitHub Actions expressions too.

**Pull request merged**